lib/googleauth/service_account.rb

```diff
@@ -43,6 +44,8 @@ module Auth
     #
     # cf [Application Default Credentials](http://goo.gl/mkAHpZ)
     class ServiceAccountCredentials < Signet::OAuth2::Client
```

Comment:
Concerns where raised about having a single type magically switch between the two very different authentication modes, particularly as the IDToken is not strictly OAuth2. While the issue is not closed a safe bet would be to make a separate class for IDToken credentials. For Java I'm planning to call this ServiceAccountIdTokenCredentials.